config/region.go

```diff
@@ -49,9 +49,12 @@ type GetRegionOpts struct {
 }
 
 func EnsureRegionsCache(ctx context.Context, c Config, opts GetRegionOpts) (RegionsCache, error) {
+	debug.Println("[Regions] Ensure cache is filled")
 	var regionsCache RegionsCache
 	fd, err := os.Open(c.RegionsCachePath)
-	if err == nil {
+	if err != nil {
+		debug.Printf("[Regions] Fail to open the cache: %v\n", err)
+	} else {
```

Comment:
We are more used to check `err != nil` in Go. So I'm refactoring to use this pattern here.

Reply:
Question: Why don't we use `defer fd.Close()`?
It seems more secure IMO.

Reply:
Question: Why don't we use defer fd.Close()?
It seems more secure IMO.

Reply:
I don't know. Maybe it's useless in this case to wait longer before closing the file descriptor: we decode its content then immediately close it.

Reply:
LGTM Then :ok_hand: